src/mc_tasks/ImpedanceTask.cpp

```diff
@@ -122,7 +123,22 @@ void ImpedanceTask::update(mc_solver::QPSolver & solver)
     deltaCompPoseW_.rotation() = aaDeltaCompRot.toRotationMatrix();
   }
 
-  // 4. Set compliance values to the targets of SurfaceTransformTask
+  // 4. Update deltaCompPoseW_ in hold mode
+  if(hold_)
+  {
+    // holdOffsetPose_ is the inverse transformation of targetPoseW_ * prevTargetPoseW_.inv()
+    holdOffsetPose_ = prevTargetPoseW_ * targetPoseW_.inv();
+    // Transform to target pose frame (see compliancePose implementation)
+    sva::PTransformd T_0_d(targetPoseW_.rotation());
+    deltaCompPoseW_ = deltaCompPoseW_ * T_0_d.inv() * holdOffsetPose_ * T_0_d;
+  }
+  else
+  {
+    holdOffsetPose_ = sva::PTransformd::Identity();
```

Comment:
This doesn't need to get updated all the time as it's irrelevant when `hold_` is false

Reply:
This is to make the `holdOffsetPose_` log easier to see (prevents the non-Identity offset from being saved when `hold_` is disabled). If you think it is still better to erase it, I'll erase it.

Reply:
Ok, maybe this can be done in `hold(bool)` then?

Reply:
Yes, it seems to be better.

In any case, `holdOffsetPose_` is no longer needed in the new hold strategy (https://github.com/jrl-umi3218/mc_rtc/pull/143#issuecomment-813045026).